### ISSUE TITLE
Unsupported namespace with includes

### ISSUE THREAD

**Issue by stephanemagnenat** [planus-org/planus#198]
Given the two files:

### `math.fbs`
```
namespace MyProject.Math;

struct Vector2 {
        x: float;
        y: float;
}
```

### `game.fbs`
```
include "math.fbs";

namespace MyProject.Game;

table GameObject {
        position: Math.Vector2;
}
```

Using google's flatbuffers works as expected (using flatc version 2.0.6):
`flatc -o rust -r --gen-all game.fbs`

However, planus is not able to resolve namespaces:
`planus check *.fbs`

```
error: Unknown type
  ┌─ game.fbs:6:12
  │
6 │     position: Math.Vector2;
  │   
```

**Comment by TethysSvensson** [comment]
Honestly I am very surprising that this compiles using `flatc`. It certainly does not seem to work though. If I try to add `game_generated.rs` and `math_generated.rs` to a project, I get name resolution errors:

```
error[E0433]: failed to resolve: could not find `math` in `super`
  --> src/game_generated.rs:66:47
   |
66 |   pub fn position(&self) -> Option<&'a super::math::Vector2> {
   |                                               ^^^^ could not find `math` in `super`
```

**Comment by TethysSvensson** [comment]
Though perhaps that is a problem with `flatc 22.11.23` specifically. I can try with a newer flatc and see what happens.

**Comment by TethysSvensson** [comment]
To use this with planus, replace `position: Math.Vector2` with `position: MyProject.Game`.

I have tried to use a newer `flatc` unsuccessfully with or without this replacement using the rust backend. As far as I can tell they don't support generating correct code for this type of nested namespaces.

**Comment by stephanemagnenat** [comment]
I will investigate more, we are using such nested namespaces for years on our project with both Typescript and Rust versions of flatbuffers.

**Comment by TethysSvensson** [comment]
Ahh, I missed the `--gen-all` flag, which seems to be required for this to work.

In any casse, I am still very confused about why `flatc` would accept this.

The resolution code in planus is [here](https://github.com/planus-org/planus/blob/59cc30e4ae44feca78abe0011c28eef1f9c0412b/crates/planus-translation/src/intermediate_language/translation.rs#L161). We currently only support two resolution modes: Relative to current namespace, and relative to the root namespace. I personally think this makes the most sense, but even so I think there is value in following the behavior of `flatc`.

I would definitely accept a patch to bring us closer to `flatc`, assuming the behavior of `flatc` is actually intentional and not just a bug.

**Comment by TethysSvensson** [comment]
@stephanemagnenat Feel free to ping my in our discord if you want to chat more about this.

**Comment by stephanemagnenat** [comment]
I did not find a specification about namespace resolution, and I cannot recall how I chose the model we use. However, a [discussion in an issue](https://github.com/google/flatbuffers/issues/5234) seems to indicate that namespace resolution uses a notion of closest distance while walking namespaces backwards, "similar to how C++ works", which seems to indicate that our usage is "correct" and not the exploitation of a bug ;-)

**Comment by TethysSvensson** [comment]
Nice find!

Unfortunately I don't have the capacity to work on this myself, but I will definitely accept a patch to make our resolution code closer to the `flatc` behavior.

**Comment by stephanemagnenat** [comment]
Of course, that's kind of an arcane feature. For my experiments with planus, I will simplify our namespace structure.

**Comment by stephanemagnenat** [comment]
Maybe there could be a "conformity" tag for such issues?
